bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -2792,24 +2806,22 @@ bool BCStateTran::getNextFullBlock(uint64_t requiredBlock,
 
   auto it = pendingItemDataMsgs.begin();
   while ((it != pendingItemDataMsgs.end()) && ((*it)->blockNumber == requiredBlock)) {
-    ItemDataMsg *msg = *it;
-
     // the conditions of these asserts are checked when receiving the message
-    ConcordAssertGT(msg->totalNumberOfChunksInBlock, 0);
-    ConcordAssertGE(msg->chunkNumber, 1);
-    if (totalNumberOfChunks == 0) totalNumberOfChunks = msg->totalNumberOfChunksInBlock;
-    blockSize += (msg->dataSize - msg->rvbDigestsSize);
-    if (totalNumberOfChunks != msg->totalNumberOfChunksInBlock || msg->chunkNumber > totalNumberOfChunks ||
+    ConcordAssertGT((*it)->totalNumberOfChunksInBlock, 0);
+    ConcordAssertGE((*it)->chunkNumber, 1);
+    if (totalNumberOfChunks == 0) totalNumberOfChunks = (*it)->totalNumberOfChunksInBlock;
+    blockSize += ((*it)->dataSize - (*it)->rvbDigestsSize);
+    if (totalNumberOfChunks != (*it)->totalNumberOfChunksInBlock || (*it)->chunkNumber > totalNumberOfChunks ||
         blockSize > maxSize) {
       badData = true;
       break;
     }
 
-    if (maxAvailableChunk + 1 < msg->chunkNumber) {
+    if (maxAvailableChunk + 1 < (*it)->chunkNumber) {
       break;  // we have a hole
     }
-    ConcordAssertEQ(maxAvailableChunk + 1, msg->chunkNumber);
-    maxAvailableChunk = msg->chunkNumber;
+    ConcordAssertEQ(maxAvailableChunk + 1, (*it)->chunkNumber);
```

Comment:
maybe extract all of the assertions to a function? 
It'll be much more readable